state/store.go

```diff
@@ -417,16 +421,24 @@ func (store dbStore) PruneStates(from int64, to int64, evidenceThresholdHeight i
 
 	err = batch.WriteSync()
 	if err != nil {
-		return err
+		return pruned, err
 	}
 
-	return nil
+	// We do not want to panic or interrupt consensus on compaction failure
+	if store.StoreOptions.Compact && previosulyPrunedStates+pruned >= uint64(store.StoreOptions.CompactionInterval) {
```

Comment:
I think there's something here that doesn't add up.
* `previosulyPrunedStates` is passed as a parameter
* There's only one callsite, and `previosulyPrunedStates` is assigned field `prunedStates` of the pruner
* So, successive calls to this method (`PruneStates`) will correctly accumulate the number of pruned states over time
* HOWEVER, when we reach `CompactionInterval`, and this if is hit, we will call compact, but field `prunedStates` of the pruner is not reset
* So... once `CompactionInterval` this IF will be hit every time... calling `store.db.Compact(nil, nil)` in every call to this method

... or am I missing something?